### PR TITLE
reboot the Scala FAQ

### PR DESCRIPTION
**Pull request by SethTisue**
gah, nobody has touched this thing in years. and, perhaps, because it's so neglected, nobody in our chat rooms and forums ever refers anyone here. (I exaggerate a bit, but it's mostly true IMO)

but people probably still find it through Google. and if we make it good, then perhaps people in chat rooms and forums _will_ refer people here more

so let's do something about it.  changes so far:

* don't duplicate content from Stack Overflow; this was a worthy but IMO failed experiment
* discourage long writeups; encourage links to answers elsewhere
* add a few general questions at the top that I know to be especially frequently asked
* drop the question about `breakOut`, it is not an FAQ these days

this PR is just a start, much room for improvement remains, but IMO this is already mergeable

The "More questions" thing is admittedly awkward -- it's the only answer in the old FAQ that wasn't just a duplicate of something from SO. Doing something about that could be future work.

### CODE REVIEW

**Comment by SethTisue** [comment]
Screen Shot 2020-12-10 at 8 44 09 AM
Screen Shot 2020-12-10 at 8 44 12 AM
Screen Shot 2020-12-10 at 8 44 15 AM

**Comment by Philippus** [comment]
This would close https://github.com/scala/docs.scala-lang/issues/1495 I think. :)

**Review by eed3si9n — APPROVED** [review]

**Comment by SethTisue** [comment]
If anyone has a suggestion for an especially obvious and important addition to the "General questions" section, you can suggest it here, and if I like your idea, I will probably add it and write an answer right now.

If your suggestion is for the "Specific technical questions" section, I would prefer that you wait and send your own PR after this one is merged.

File: _overviews/FAQ/index.md [hunk 1]
```diff
@@ -4,10 +4,84 @@ title: Scala FAQs
...
+
+### What are Scala context bounds (`[T : Foo]`)?
+
+[answer on Stack Overflow](https://stackoverflow.com/a/4467012)
+
+### How does `yield` work?
+
+[answer on Stack Overflow](https://stackoverflow.com/a/1059501)
+
+### What is the difference between view, stream and iterator?
```
Comment (sjrd) [inline _overviews/FAQ/index.md:72]: Should this mention `LazyList` now?

**Review by sjrd — APPROVED** [review]

File: _overviews/FAQ/index.md (hunk 1 above)
Comment (SethTisue) [reply _overviews/FAQ/index.md:72]: yeah. hopefully somebody (perhaps me? idk) will choose to update that in a followup PR

**Comment by SethTisue** [comment]
> this PR is just a start, much room for improvement remains, but IMO this is already mergeable

My rough plan is:

1) merge this PR, but only publicize it to contributors for now
2) hope some volunteers materialize to get it in even better shape
3) *then* publicize it more broadly

**Comment by SethTisue** [comment]
(before we merge this, I need to add a commit that adds redirects for the removed pages)

**Comment by SethTisue** [comment]
related: https://github.com/MasseGuillaume/sca.la/pull/59 , since the FAQ's URL is hard to remember

**Comment by SethTisue** [comment]
> before we merge this, I need to add a commit that adds redirects for the removed pages

haha well, I'd better go write that commit right now then :-)

**Comment by SethTisue** [comment]
https://github.com/scala/docs.scala-lang/pull/1845

**Comment by SethTisue** [comment]
more followup: #1846

**Pull request merged**